scintellometry/io/arochime.py

```diff
@@ -167,6 +168,85 @@ def open(self, number=0):
         return self.fh_raw
 
 
+class AROCHIMEInvPFB(SequentialFile):
+
+    telescope = 'arochime-invpfb'
+
+    def __init__(self, raw_files, blocksize, samplerate, fedge, fedge_at_top,
+                 time_offset=0.0*u.s, dtype='4bit,4bit', comm=None):
+        """ARO data acquired with a CHIME correlator, PFB inverted.
+        """
+        self.fh_raw = AROCHIMERawData(raw_files, blocksize, samplerate,
+                                      fedge, fedge_at_top, time_offset,
+                                      dtype='cu4bit,cu4bit', comm=comm)
+        self.time0 = self.fh_raw.time0
+        self.samplerate = self.fh_raw.samplerate
+        self.dtsample = (1 / self.samplerate).to(u.s)
+        self.npol = self.fh_raw.npol
+        self.fedge = self.fh_raw.fedge
+        self.fedge_at_top = self.fh_raw.fedge_at_top
+        super(AROCHIMEInvPFB, self).__init__(raw_files, blocksize, dtype,
+                                             nchan=1, comm=comm)
+        # PFB information
+        self.nblock = 2048
+        self.h = pfb.sinc_hamming(4, self.nblock).reshape(4, -1)
+        self.fh = None
+        # S/N for use in the Wiener Filter
+        prec = (1 << 3) ** 0.5
+        self.sn = prec / 0.2887
+        print("Opened InvPFB reader, S/N={0}".format(self.sn))
+
+    def open(self, number=0):
+        self.fh_raw.open(number)
+
+    def close(self):
+        self.fh_raw.close()
+
+    def _seek(self, offset):
+        if offset % self.recordsize != 0:
+            raise ValueError("Cannot offset to non-integer number of records")
+        self.offset = (offset // self.fh_raw.recordsize *
+                       self.fh_raw.recordsize)
+
+    def seek_record_read(self, offset, size):
+        print('Inverting PFB... ')
```

Comment:
Could you add a comment or docstring here, which described what is done, and also notes that with the present implementation, there are some edge effects? (i.e., that to do it right, one would need to read in more than requested).